src/useTitle.ts

```diff
@@ -1,11 +1,14 @@
 import { useRef } from 'react';
 
 function useTitle(title: string) {
```

Comment:
I think upgraded `useTitle` can be like:

```suggestion
import { useEffect, useRef } from 'react';

type Options = {
  restoreOnUnmount?: boolean;
};

function useTitle(
  title: string,
  { restoreOnUnmount = false }: Options = {}
) {
  const prevTitle = useRef(document.title);

  document.title = title;

  useEffect(() => {
    if (restoreOnUnmount) {
      return () => {
        document.title = prevTitle.current;
      };
    }
  }, []);
}

```

The main point in fact that restoration must be optional

Reply:
@zsh2401 can we make title restoration on un-mount optional, like @FlickerLogicalStack proposed, so this change is not breaking?

Reply:
@streamich I DO AGREE! I will do it as soon as possible!